conans/assets/templates/new_v2_bazel.py

```diff
@@ -24,28 +26,28 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
-    def validate(self):
-        if self.settings.os in ("Windows", "Macos") and self.options.shared:
-            raise ConanException("Windows and Macos needs extra BUILD configuration to be able "
-                                 "to create a shared library. Please, check this reference to "
-                                 "know more about it: https://bazel.build/reference/be/c-cpp")
-
     def layout(self):
-        bazel_layout(self)
         # DEPRECATED: Default generators folder will be "conan" in Conan 2.x
         self.folders.generators = "conan"
+        bazel_layout(self)
 
     def build(self):
         bazel = Bazel(self)
-        bazel.build()
+        # Linux creates both shared and static libraries by default, and
+        # it is even conflicting if we use the cc_shared_library rule
+        if self.options.shared and self.settings.os != "Linux":
+            # main/BUILD uses cc_shared_library to create shared libraries
```

Comment:
Maybe a more explicit comment? Something like this?
```suggestion
            # We need to add '--experimental_cc_shared_library' because the project uses cc_shared_library to create shared libraries
```